### PR TITLE
VoiceMessages: Support replies

### PR DESCRIPTION
**Pull request by nmsturcke**
Add a reply function to the voice message plugin.

### CODE REVIEW

File: src/plugins/voiceMessages/index.tsx [hunk 1]
```diff
@@ -37,6 +37,8 @@ import { VoicePreview } from "./VoicePreview";
 import { VoiceRecorderWeb } from "./WebRecorder";
 
 const CloudUpload = findLazy(m => m.prototype?.uploadFileToCloud);
+const MessageCreator = findByPropsLazy("getSendMessageOptionsForReply", "sendMessage");
+const PendingReplyStore = findByPropsLazy("getPendingReply");
```
Comment (TheKodeToad) [inline src/plugins/voiceMessages/index.tsx:41]:
```suggestion
const PendingReplyStore = Vencord.Webpack.findStoreLazy("PendingReplyStore");
```

Comment (Vap0r1ze) [reply src/plugins/voiceMessages/index.tsx:41]: This but you should just `import { findStoreLazy } from "@webpack";`

Comment (TheKodeToad) [reply src/plugins/voiceMessages/index.tsx:41]: I copied it from DevTools :woman_facepalming:

File: src/plugins/voiceMessages/index.tsx [hunk 2]
```diff
@@ -71,6 +73,8 @@ const EMPTY_META: AudioMetadata = {
 
 function sendAudio(blob: Blob, meta: AudioMetadata) {
     const channelId = SelectedChannelStore.getChannelId();
+    const reply = PendingReplyStore.getPendingReply(channelId);
+    (reply && FluxDispatcher.dispatch({ type: "DELETE_PENDING_REPLY", channelId }));
```
Comment (Vap0r1ze) [inline src/plugins/voiceMessages/index.tsx:77]:
```suggestion
    if (reply) FluxDispatcher.dispatch({ type: "DELETE_PENDING_REPLY", channelId });
```
just a little more readable

Comment (TheKodeToad) [reply src/plugins/voiceMessages/index.tsx:77]: why not
```suggestion
    if (reply)
       FluxDispatcher.dispatch({ type: "DELETE_PENDING_REPLY", channelId });
```

File: src/plugins/voiceMessages/index.tsx (hunk 1 above)
Comment (nmsturcke) [reply src/plugins/voiceMessages/index.tsx:41]: I just copied the spotify share plugin for pending reply lol

File: src/plugins/voiceMessages/index.tsx [hunk 3]
```diff
@@ -95,7 +99,8 @@ function sendAudio(blob: Blob, meta: AudioMetadata) {
                     uploaded_filename: upload.uploadedFilename,
                     waveform: meta.waveform,
                     duration_secs: meta.duration,
-                }]
+                }],
+                message_reference: reply ? MessageCreator.getSendMessageOptionsForReply(reply).messageReference : null,
```
Comment (Vendicated) [inline src/plugins/voiceMessages/index.tsx:103]:
```suggestion
                message_reference: reply ? MessageCreator.getSendMessageOptionsForReply(reply)?.messageReference : null,
```

**Pull request merged**